ocaml/xapi/livepatch.ml

```diff
@@ -187,43 +187,46 @@ module KernelLivePatch = struct
 end
 
 module XenLivePatch = struct
-  let get_regexp status =
-    Re.Posix.compile_pat
-      (Printf.sprintf {|^[ ]*lp_([^- ]+)-([^- ]+)-([^- ]+)-([^- ]+).+%s.*$|}
-         status
-      )
+  let drop x =
+    let open Astring.Char.Ascii in
+    is_control x || is_blank x || is_white x
 
-  let get_livepatches pattern s =
+  let get_livepatches state s =
+    let pattern =
+      Re.Posix.compile_pat {|^lp_([^- ]+)-([^- ]+)-([^- ]+)-([^- ]+)$|}
```

Comment:
Unfortunately, the content in the list is unpredictable. So the regexp is trying to match as many as possible so that the following command line can report error if the matched things are wrong indeed. The un-matching will be ignored silently.